pydiggy/node.py

```diff
@@ -136,24 +123,42 @@ def __new__(cls, name, bases, attrs, **kwargs):
 
 
 class Node(metaclass=NodeMeta):
-    uid: int
+    _instances = dict()
+    _staged = dict()
+
+
+    @classmethod
+    def _get_staged(cls):
+        return cls._staged
+
+    @classmethod
+    def _clear_staged(cls):
+        cls._staged = {}
+        # NodeTypeRegistry._i needs to be updated when clearing staged nodes
 
-    _i = _count()
-    _nodes = []
-    _staged = {}
+    @classmethod
+    def _reset(cls) -> None:
+        cls._instances = dict()
+
+    @classmethod
+    def _get_name(cls) -> str:
+        return cls.__name__
 
     def __init_subclass__(cls, is_abstract: bool = False) -> None:
         if not is_abstract:
-            cls._register_node(cls)
+            from pydiggy.node_type_registry import NodeTypeRegistry

```

Comment:
Why the import inside the method?

Reply:
Avoiding circular imports. Short of doing more extensive refactoring, putting imports inside of implementations is the way to do it. 

I think that the only way to avoid this would be to have NodeTypeRegistry be in the same file as Node, or to remove the auto-registration.

Reply:
Got it. I would rather them be in the same file than nesting the imports. And, I see the auto-registration as a necessary feature. Maybe it can be turned off in the future, but it makes the end API super simple and intuitive to use (which was the goal).